blobs.go

```diff
@@ -238,3 +244,22 @@ func isBinary(blob *object.Blob) (bool, error) {
 		}
 	}
 }
+
+func shouldReadContent(columns []sql.Expression) bool {
```

Comment:
This maybe should be a rule that adds a boolean to blobs table?

Reply:
I think the rule will end up being more expensive than just walking a few expressions. This is done once when the iterator is created, the rule will probably trigger a new pass of the analyzer (since the analyzer creates new nodes and the tree will change) and all its rules, which may end up being way more expensive than checking this here.

Note that this part is only used for non-squash. In squash, it's done in the rule because iterators are built on analysis time.